### ISSUE TITLE
Unsatisfiable constraints: php5-mcrypt (missing) building docker image

### ISSUE THREAD

**Issue by jgautsch** [handnot2/samly_simplesaml#3]
The error seems to be fixed by explicitly using an older version of the `php:fpm-alpine` image at the top of the Dockerfile.

Simple Dockerfile change fixes this problem:
```diff
- FROM php:fpm-alpine
+ FROM php:5.6.31-fpm-alpine
```

There may be a better explicit version to use, and probably even better- a fix that uses the latest version. Sorry for not opening a PR, I'm just plowing along at the moment.

Thank you for creating the samly lib!

Full error:

```bash
(master) $ docker --debug=true build -t "samlysaml" -f Dockerfile . --no-cache
DEBU[0002] running session: uo75pd7z5gfd53qvwqdeenvtl
DEBU[0002] serving grpc connection
Sending build context to Docker daemon  227.3kB
Step 1/7 : FROM php:fpm-alpine
fpm-alpine: Pulling from library/php
6c40cc604d8e: Already exists
a4af09669d6b: Pull complete
ba64579e88cb: Pull complete
1d54458d46b3: Pull complete
83b3ebf15fbb: Pull complete
bdfd02670318: Pull complete
19f2ed82081d: Pull complete
1df93195e010: Pull complete
e9b3f5c5edee: Pull complete
720ec860b268: Pull complete
Digest: sha256:69faa5d59f1e061dd16600b63c54e195f8c7054049dee6d3959b08953247c557
Status: Downloaded newer image for php:fpm-alpine
 ---> b51832111996
Step 2/7 : ENV SAML_VERSION 1.16.2
 ---> Running in c0fd6023977c
Removing intermediate container c0fd6023977c
 ---> 6ea2622a82ea
Step 3/7 : RUN set -xe   && apk add --no-cache openssl   && apk add --no-cache php5-mcrypt   && apk add --no-cache php5-ldap 	&& buildDeps="wget ca-certificates" 	&& apk add --no-cache $buildDeps 	&& update-ca-certificates 	&& rm -rf /var/cache/apk/* 	&& wget https://github.com/simplesamlphp/simplesamlphp/releases/download/v${SAML_VERSION}/simplesamlphp-${SAML_VERSION}.tar.gz 		-O /tmp/simplesamlphp-${SAML_VERSION}.tar.gz --no-check-certificate 	&& tar -zxf /tmp/simplesamlphp-${SAML_VERSION}.tar.gz -C /tmp 	&& mv /tmp/simplesamlphp-${SAML_VERSION} /srv/simplesaml 	&& wget https://github.com/bander2/twit/releases/download/1.1.0/twit-linux-amd64 		-O /usr/local/bin/twit --secure-protocol=TLSv1_2 	&& chmod u+x /usr/local/bin/twit 	&& mkdir /lib64 	&& ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2 	&& apk del $buildDeps
 ---> Running in 601070f4b012
+ apk add --no-cache openssl
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
OK: 18 MiB in 32 packages
+ apk add --no-cache php5-mcrypt
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  php5-mcrypt (missing):
    required by: world[php5-mcrypt]
The command '/bin/sh -c set -xe   && apk add --no-cache openssl   && apk add --no-cache php5-mcrypt   && apk add --no-cache php5-ldap 	&& buildDeps="wget ca-certificates" 	&& apk add --no-cache $buildDeps 	&& update-ca-certificates 	&& rm -rf /var/cache/apk/* 	&& wget https://github.com/simplesamlphp/simplesamlphp/releases/download/v${SAML_VERSION}/simplesamlphp-${SAML_VERSION}.tar.gz 		-O /tmp/simplesamlphp-${SAML_VERSION}.tar.gz --no-check-certificate 	&& tar -zxf /tmp/simplesamlphp-${SAML_VERSION}.tar.gz -C /tmp 	&& mv /tmp/simplesamlphp-${SAML_VERSION} /srv/simplesaml 	&& wget https://github.com/bander2/twit/releases/download/1.1.0/twit-linux-amd64 		-O /usr/local/bin/twit --secure-protocol=TLSv1_2 	&& chmod u+x /usr/local/bin/twit 	&& mkdir /lib64 	&& ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2 	&& apk del $buildDeps' returned a non-zero code: 1
```

**Comment by handnot2** [comment]
Yes. You are right. It is better to use explicit version number. Let me check and get back on the correct version.

**Comment by handnot2** [comment]
Locally I have been using `php:7.1-fpm-alpine3.8`. I will fix this tomorrow.

**Comment by jgautsch** [comment]
No rush on my account!
Thanks so much for the great samly library!

**Comment by handnot2** [comment]
Addressed in v0.4 along with changes to how multiple IdP instanced can be managed.
Reopen if you still have any issues.
